api/v1alpha1/identityproviderconfiguration_types.go

```diff
@@ -0,0 +1,94 @@
+package v1alpha1
+
+import (
+	lifecycleapi "github.com/platform-mesh/golang-commons/controller/lifecycle/api"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type IdentityProviderClientType string
+
+const (
+	IdentityProviderClientTypeConfidential IdentityProviderClientType = "confidential"
+	IdentityProviderClientTypePublic       IdentityProviderClientType = "public"
+)
+
+type IdentityProviderClientConfig struct {
+	// +kubebuilder:validation:Enum=confidential;public
+	ClientType                  IdentityProviderClientType `json:"clientType"`
+	ClientID                    string                     `json:"clientID"`
+	ClientName                  string                     `json:"clientName"`
+	ValidRedirectUris           []string                   `json:"validRedirectUris"`
+	ValidPostLogoutRedirectUris []string                   `json:"validPostLogoutRedirectUris,omitempty"`
+	// +optional
+	ClientSecretRef       ClientSecretRef `json:"clientSecretRef"`
+	RegistrationClientURI string          `json:"registrationClientURI,omitempty"`
+}
+
+type ClientSecretRef struct {
+	corev1.SecretReference `json:",inline"`
+}
+
+// IdentityProviderConfigurationSpec defines the desired state of IdentityProviderConfiguration
+type IdentityProviderConfigurationSpec struct {
+	Clients []IdentityProviderClientConfig `json:"clients"`
+}
+
+// ManagedClient tracks a client that is managed by the operator.
+type ManagedClient struct {
+	ClientID              string          `json:"clientID"`
```

Comment:
```suggestion
	ClientID              string          `json:"clientId"`
```

same comment as above